cmd/proxy/actions/filter.go

```diff
@@ -1,24 +1,32 @@
 package actions
 
 import (
+	"bytes"
+	"encoding/json"
 	"net/http"
 	"net/url"
 	"strings"
 
 	"github.com/bketelsen/buffet"
 	"github.com/gobuffalo/buffalo"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/download"
 	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/log"
 	"github.com/gomods/athens/pkg/module"
 	"github.com/gomods/athens/pkg/paths"
 )
 
-func newFilterMiddleware(mf *module.Filter) buffalo.MiddlewareFunc {
+func newFilterMiddleware(mf *module.Filter, lggr *log.Logger) buffalo.MiddlewareFunc {
 	const op errors.Op = "actions.FilterMiddleware"
 
 	return func(next buffalo.Handler) buffalo.Handler {
 		return func(c buffalo.Context) error {
 			sp := buffet.SpanFromContext(c).SetOperationName("filterMiddleware")
 			defer sp.Finish()
+
+			entry := download.ContextLogEntry(c, lggr)
```

Comment:
the `download` package is the wrong place to instantiate a log Entry from *log.Logger. I know it does that for its own handlers but that's potentially a special case for type safety. If we want to merge them, then we should create a middleware package where we put all of our middleware handlers there and instantiate a request-scoped Entry in its own middleware. We don't have to merge them right away, but it's up to you